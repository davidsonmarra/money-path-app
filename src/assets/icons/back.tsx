import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BackIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-back">
    <Path d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z" />
  </Svg>
);

export default BackIcon;
