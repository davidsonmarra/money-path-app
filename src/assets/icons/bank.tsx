import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BankIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-bank">
    <Path d="M200-280v-280h80v280zm240 0v-280h80v280zM80-120v-80h800v80zm600-160v-280h80v280zM80-640v-80l400-200 400 200v80zm178-80h444zm0 0h444L480-830z" />
  </Svg>
);

export default BankIcon;
