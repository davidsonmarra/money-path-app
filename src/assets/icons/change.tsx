import React from 'react';
import Svg, { Path } from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const ChangeIcon = ({ size = 24, color = '#FFF' }: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-change"
  >
    <Path d="m482-200 114-113-114-113-42 42 43 43q-28 1-54.5-9T381-381q-20-20-30.5-46T340-479q0-17 4.5-34t12.5-33l-44-44q-17 25-25 53t-8 57q0 38 15 75t44 66 65 43.5 74 15.5l-38 38zm165-170q17-25 25-53t8-57q0-38-14.5-75.5T622-622t-65.5-43-74.5-14l38-39-42-42-114 113 114 113 42-42-44-44q27 0 55 10.5t48 30.5 30.5 46 10.5 52q0 17-4.5 34T603-414zM480-80q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default ChangeIcon;
