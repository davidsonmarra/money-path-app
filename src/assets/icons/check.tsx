import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const CheckIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-check">
    <Path d="m424-296 282-282-56-56-226 226-114-114-56 56zm56 216q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default CheckIcon;
