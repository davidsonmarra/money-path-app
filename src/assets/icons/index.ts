export {default as AppleIcon} from 'src/assets/icons/apple';
export {default as BackIcon} from 'src/assets/icons/back';
export {default as BancoDoBrasilIcon} from 'src/assets/icons/banco-do-brasil';
export {default as BankIcon} from 'src/assets/icons/bank';
export {default as CheckIcon} from 'src/assets/icons/check';
export {default as ChevronRightIcon} from 'src/assets/icons/chevron-right';
export {default as CloseIcon} from 'src/assets/icons/close';
export {default as ErrorIcon} from 'src/assets/icons/error';
export {default as GoogleIcon} from 'src/assets/icons/google';
export {default as InterIcon} from 'src/assets/icons/inter';
export {default as ItauIcon} from 'src/assets/icons/itau';
export {default as MoneyPathIcon} from 'src/assets/icons/money-path';
export {default as NubankIcon} from 'src/assets/icons/nubank';
export {default as SantanderIcon} from 'src/assets/icons/santander';
export {default as WalletIcon} from 'src/assets/icons/wallet';
export {default as WarningIcon} from 'src/assets/icons/warning';
