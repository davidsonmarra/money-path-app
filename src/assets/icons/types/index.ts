export enum IconType {
  apple = 'apple',
  back = 'back',
  bancoDoBrasil = 'bancoDoBrasil',
  bank = 'bank',
  check = 'check',
  chevronRight = 'chevronRight',
  close = 'close',
  error = 'error',
  food = 'food',
  games = 'games',
  google = 'google',
  health = 'health',
  help = 'help',
  house = 'house',
  inter = 'inter',
  itau = 'itau',
  moneyPath = 'moneyPath',
  money = 'money',
  nubank = 'nubank',
  santander = 'santander',
  transport = 'transport',
  wallet = 'wallet',
  warning = 'warning',
}
