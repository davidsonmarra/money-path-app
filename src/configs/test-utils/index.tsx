import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
