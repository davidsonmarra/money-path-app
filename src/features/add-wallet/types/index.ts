import {IconType} from 'src/assets/icons/types';

export interface WalletProps {
  name: string;
  icon: IconType;
  color?: string;
  backgroundColor?: string;
  type: 'personal' | 'bank';
  amount?: number;
}

export type SelectWalletSection = {
  title: string;
  data: WalletProps[];
};
