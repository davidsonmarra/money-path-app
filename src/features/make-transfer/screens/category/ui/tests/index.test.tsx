import React from 'react';
import { fireEvent, render } from 'src/configs/test-utils';
import CategoryContainer from 'src/features/make-transfer/screens/category/ui';
import { IconType } from 'src/assets/icons/types';
import Category from 'src/@types/category';

const mockCategories: Category[] = [
  {
    id: '1',
    name: 'Alimentação',
    backgroundColor: '#FF6B35',
    color: '#FFFFFF',
    icon: IconType.food,
  },
  {
    id: '2',
    name: 'Transporte',
    backgroundColor: '#4A90E2',
    color: '#FFFFFF',
    icon: IconType.transport,
  },
];

const containerInstance = (props: any) =>
  render(<CategoryContainer {...props} />);

describe('CategoryContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with default status', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loading status', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'loading',
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with error status', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'error',
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call onBack when header back button is pressed', () => {
    const mockOnBack = jest.fn();
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
      onBack: mockOnBack,
    });

    container.getByTestId('btn-left-icon').props.onClick();
    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });

  it('should call setCategory when a category is selected', () => {
    const mockSetCategory = jest.fn();
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
      setCategory: mockSetCategory,
    });

    fireEvent.press(container.getByText('Alimentação'));
    expect(mockSetCategory).toHaveBeenCalledWith('1');
  });

  it('should call onConfirm when confirm button is pressed', () => {
    const mockOnConfirm = jest.fn();
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
      onConfirm: mockOnConfirm,
      selectedCategoryId: '1',
    });

    fireEvent.press(container.getByTestId('btn-confirm'));
    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
  });

  it('should show loading spinner when status is loading', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'loading',
    });

    expect(container.getByTestId('loading-spinner')).toBeTruthy();
  });

  it('should show error message when status is error', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'error',
    });

    expect(container.getByText('Erro ao carregar categorias')).toBeTruthy();
  });

  it('should render all categories correctly', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
    });

    expect(container.getByText('Alimentação')).toBeTruthy();
    expect(container.getByText('Transporte')).toBeTruthy();
  });

  it('should apply selected style to selected category', () => {
    const container = containerInstance({
      categories: mockCategories,
      status: 'default',
      selectedCategoryId: '1',
    });

    const alimentacaoItem = container.getByText('Alimentação').parent;
    expect(alimentacaoItem).toBeTruthy();
  });
});
