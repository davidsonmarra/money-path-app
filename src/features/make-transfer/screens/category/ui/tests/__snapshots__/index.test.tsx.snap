// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryContainer should render correctly with default status 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Categoria
    </Text>
    <View />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RNCSafeAreaView
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      edges={
        {
          "bottom": "additive",
          "left": "off",
          "right": "off",
          "top": "off",
        }
      }
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#232E38",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <View>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 20,
                "fontWeight": 600,
                "lineHeight": 24,
              },
              undefined,
            ]
          }
        >
          Selecione uma categoria
        </Text>
        <View
          style={
            {
              "height": 12,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontWeight": 400,
                },
              ],
              undefined,
            ]
          }
        >
          Para criar uma transferência, precisamos que você selecione uma categoria ou crie uma nova:
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 12,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#232E38",
              "borderRadius": 12,
              "opacity": 0.5,
              "padding": 12,
              "width": "30%",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 32,
                  "height": 64,
                  "justifyContent": "center",
                  "width": 64,
                },
                {
                  "backgroundColor": "#FF6B35",
                },
              ]
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={40}
              bbWidth={40}
              fill="#FFFFFF"
              focusable={false}
              height={40}
              meetOrSlice={0}
              minX={0}
              minY={-960}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
              testID="icon-food"
              vbHeight={960}
              vbWidth={960}
              width={40}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4294967295,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M280-80v-366q-51-14-85.5-56T160-600v-280h80v280h40v-280h80v280h40v-280h80v280q0 56-34.5 98T360-446v366zm400 0v-320H560v-280q0-83 58.5-141.5T760-880v800z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "height": 8,
                "width": 0,
              }
            }
            testID="spacer"
          />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontWeight": 400,
                  },
                ],
                {
                  "textAlign": "center",
                },
              ]
            }
          >
            Alimentação
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#232E38",
              "borderRadius": 12,
              "opacity": 0.5,
              "padding": 12,
              "width": "30%",
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 32,
                  "height": 64,
                  "justifyContent": "center",
                  "width": 64,
                },
                {
                  "backgroundColor": "#4A90E2",
                },
              ]
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={40}
              bbWidth={40}
              fill="#FFFFFF"
              focusable={false}
              height={40}
              meetOrSlice={0}
              minX={0}
              minY={-960}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
              testID="icon-transport"
              vbHeight={960}
              vbWidth={960}
              width={40}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4294967295,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M240-200v40q0 17-11.5 28.5T200-120h-40q-17 0-28.5-11.5T120-160v-320l84-240q6-18 21.5-29t34.5-11h440q19 0 34.5 11t21.5 29l84 240v320q0 17-11.5 28.5T800-120h-40q-17 0-28.5-11.5T720-160v-40zm-8-360h496l-42-120H274zm-32 80v200zm100 160q25 0 42.5-17.5T360-380t-17.5-42.5T300-440t-42.5 17.5T240-380t17.5 42.5T300-320m360 0q25 0 42.5-17.5T720-380t-17.5-42.5T660-440t-42.5 17.5T600-380t17.5 42.5T660-320m-460 40h560v-200H200z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "height": 8,
                "width": 0,
              }
            }
            testID="spacer"
          />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontWeight": 400,
                  },
                ],
                {
                  "textAlign": "center",
                },
              ]
            }
          >
            Transporte
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#4F4F4F",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 24,
              "paddingVertical": 12,
              "width": "100%",
            }
          }
          testID="btn-confirm"
        >
          <View />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  {
                    "fontWeight": 600,
                  },
                ],
                {
                  "color": "#FFF",
                },
              ]
            }
          >
            Continuar
          </Text>
          <View />
        </View>
        <View
          style={
            {
              "height": 8,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 24,
              "paddingVertical": 12,
              "width": "100%",
            }
          }
          testID="btn-create-category"
        >
          <View />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  {
                    "fontWeight": 600,
                  },
                ],
                {
                  "color": "#2A5C99",
                },
              ]
            }
          >
            Criar nova categoria
          </Text>
          <View />
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</RNCSafeAreaView>
`;

exports[`CategoryContainer should render correctly with error status 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Categoria
    </Text>
    <View />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            [
              {
                "color": "#D9D9D9",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontWeight": 400,
              },
            ],
            undefined,
          ]
        }
      >
        Erro ao carregar categorias
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`CategoryContainer should render correctly with loading status 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Categoria
    </Text>
    <View />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        color="#EAC43D"
        size="large"
        testID="loading-spinner"
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
